aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -428,6 +429,24 @@ module aptos_framework::account {
         (signer, signer_cap)
     }
 
+    /// Create a resource account using the vectorized Unix timestamp as a seed.
+    ///
+    /// Simplifies resource account creation, since callers do not need to specify a seed.
+    ///
+    /// In practice the Unix timestamp is effectively non-deterministic, or more specifically, just
+    /// as difficult to front run in a miner extractable value (MEV)-style attack as a seed
+    /// provided via a `public entry` function argument. That is, if a malicious block producer
+    /// wanted to prevent the creation of a resource account, they would have to construct a block
+    /// with a carefully-selected timestamp that allows them to insert an account creation
+    /// transaction ahead of the legitimate account creation transaction. This is comparable in
+    /// difficulty to the effort required for front-running resource account creation when the seed
+    /// is supplied via a `public entry` function, and both such attacks are more difficult still
+    /// than front-running the creation of a resource account via a hard-coded seed.
+    public fun create_resource_account_using_timestamp(source: &signer): (signer, SignerCapability)
```

Comment:
This function would fail if called twice in the same block right? Do we want to add the source account's nonce here to prevent that from happening?

Reply:
@movekevin [f46ac6d5330a1d7de57fb6989c2ed60901565583](https://github.com/aptos-labs/aptos-core/pull/4173/commits/f46ac6d5330a1d7de57fb6989c2ed60901565583) adds a disclaimer about failing multiple times if called within a single block by the same `source`. Tagging @chen-robert who expressed similar concerns.

Could you clarify what you mean by the source account's nonce? I am not sure if I recognize nonce as a field, for instance in `accounts.move`, and I'll add that I don't know whether this would solve the issue of multiple calls from the same `source` within a single transaction.

Reply:
Yeah it wouldn't solve that if there are multiple calls with the same tx. I was thinking of solving the failure across multiple txs in the same block. How about allow user to pass a custom seed here as well that will get appended to the timestamp bytes? This allows users/modules that use it to have a way to do call this multiple times within the same block by passing an incrementing counter.

Reply:
I think by nonce you mean the `Account::sequence_number`, @movekevin ?

Reply:
I find this comment confusing:
1. How is the timestamp non-deterministic? It's quite a deterministically increasing number, isn't it?
2. It seems like the timestamp is **more** difficult to front-run than the seed, not just as difficult. With a seed, an attacker who sees a TXN can get the seed, predict the address of the resource account, and call `aptos_account::transfer(attacker_source, predicted_address, num_coins)`. With a timestamp, the attacker has to have many guesses of `predicted_address` based on what the timestamp will be, no?

Reply:
> 

@movekevin adding a second argument here may defeat the primary of the PR, namely, to provide a resource account generation mechanism that doesn't require an external argument. Your thoughts?

@alinush Agreed that the whole bit about quasi-nondeterminism is really just about preventing front-running, and yes you are right about an attacker who simply blocks a transaction by sniffing the seed ahead of time. The timestamp approach is more difficult to front-run i practice, but in theory you are right that a malicious actor could front-run either. @davidiw indicated yesterday that in practice this is unlikely, and it appears it is a result of system throughput in general being high enough to make such analysis impractical. As for making it impossible to front-run resource account in general, would this require re-implementing the resource address generation?

Reply:
Currently, I don't see an easy fix for front-running. I might have something in mind based on commitments, but it requires different namespaces for addresses created via `create_account` versus addresses created via `create_resource_account`.

Reply:
"@movekevin adding a second argument here may defeat the primary of the PR, namely, to provide a resource account generation mechanism that doesn't require an external argument. Your thoughts?" => Why? That second param is a vector. People can pass an empty vector to just get the default behavior (and the risk of collision if they do this multiple times in the same block)

Reply:
I don't think it's important that we solve front-running here. It's always a problem in general with create_account being public and callable by anyone.

Reply:
> 

@movekeving regarding the passing an empty vector: if this the passing of an empty vector is hard-coded in a module, then a single blockage prevents initialization without callers updating their code. And as for passing an incrementing number, this requires either passing in the incremented number at the `public entry fun` level, or allocating space in global storage for an incrementing counter, both of which impose additional work on callers. Your thoughts?

Reply:
Isn't that what I suggested the adding a second param of a vector for? If "if the passing of an empty vector is hard-coded in a module, then a single blockage prevents initialization without callers updating their code" is a concern, modules should do the extra work with the incrementing number or whatever is needed. If it's not a concern at all, empty vector works. If you want, we can really make this a lot more complex and add all the bells and whistles but for now, IMO, that is a relatively clean solution for allowing users to solve the collision problem if it's important to them

Reply:
@movekevin [f2ff2e6670eaf6e0990139dcc5c387fc6287087d](https://github.com/aptos-labs/aptos-core/pull/4173/commits/f2ff2e6670eaf6e0990139dcc5c387fc6287087d) incorporates a second parameter as you suggested, and, per game theoretic considerations discussed with @alinush and @CapCap last night, implements a handicap-principle based method of creation enforcement.